angular/src/lib/input/examples/default.component.ts

```diff
@@ -1,60 +1,351 @@
+      <textarea
+        mdInput
+        [(ngModel)]="Filledmultiline"
+        placeholder='FilledMultiline'
+      ></textarea>
+    </md-input-group>
+    <md-input-group label='Filled Read Only' [isFilled]="true" containerSize="small-12">
+      <input
+        mdInput
+        readonly
+        [(ngModel)]="Filledreadonly"
+        placeholder='Filled Read Only'
+      />
+    </md-input-group>
+    <md-input-group label='FilledDisabled' [isFilled]="true" containerSize="small-12">
+      <input
+        mdInput
+        disabled
+        [(ngModel)]="Filleddisabled"
+        placeholder='FilledDisabled'
+      />
+    </md-input-group>
+    <md-input-group label='Filled Input Before' [isFilled]="true" containerSize="small-12">
+      <input
+        mdInput
+        [(ngModel)]="FilledinputBefore"
+        placeholder='Filled Input Before'
+      />
+    </md-input-group>
+    <md-input-group label='Filled Input After' [isFilled]="true" containerSize="small-12">
+      <input
+        mdInput
+        [(ngModel)]="filledinputAfter"
+        placeholder='Filled Input After'
+      />
+    </md-input-group>
+    <md-input-group label='Filled Clear' [clear]='true' [isFilled]="true" containerSize="small-12">
+      <input
+        mdInput
+        [(ngModel)]="filledclear"
+        placeholder='Filled Clear'
+      />
+    </md-input-group>
+    <md-input-group label='Hint' inputHelpText="Filled Hint label" [isFilled]="true" containerSize="small-12">
+      <input
+        mdInput
+        [(ngModel)]="filledhint"
+        placeholder='Filled Hint'
+      />
+    </md-input-group>
   `,
 })
-export class ExampleInputDefaultComponent {
-  dataModel: string = 'Test';
 
-  inputForm;
+export class ExampleInputDefaultComponent implements OnInit {
+  warningReactive: boolean;
+  warningTemplate: boolean;
+  templateValue: string = 'Template Version';
+  inputForm: any;
 
   constructor(private fb: FormBuilder) {
-
     this.inputForm = this.fb.group({
-      inputControl: ['', Validators.compose(
-        [ Validators.required,
-          Validators.minLength(10),
-          Validators.maxLength(14),
-          ValidateCustom
-        ]),
-      ]
+      success: ['ReactiveForm Version', Validators.minLength(4)],
+      warning: ['ReactiveForm Version', this.ValidateCustom],
+      error: ['ReactiveForm Version', Validators.compose([
+        Validators.minLength(40),
+        this.ValidateCustom
+      ])]
     });
+  }
 
-    console.info(this.inputForm);
+  ngOnInit() {
+    this.warningTemplate = !this.templateValue.startsWith('warning');
+    this.error.markAsDirty();
   }
-}
 
-function ValidateCustom(control: AbstractControl) {
-  if (!control.value.startsWith('custom')) {
-    return { customValid: true };
+  get error() { return this.inputForm.get('error'); }
+
+  private ValidateCustom = (control: AbstractControl) => {
+    this.warningReactive = !control.value.startsWith('reactive');
+
+    return null;
   }
-  return null;
 }
+
+
```

Comment:
Should this example be broken up into multiple examples?  Seems like a lot of things to digest in one example.

Reply:
Yes it will be - this will be the kitchensink example when the pr is complete.